src/wagtailtrans/views/language.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import absolute_import, unicode_literals
+
+from django.db import models
+from django.shortcuts import redirect
+from django.utils.translation import ugettext_lazy as _
+from wagtail.contrib.modeladmin.views import DeleteView
+from wagtail.wagtailadmin import messages
+
+
+class LanguageDeleteView(DeleteView):
+    """
+    Custom Delete View Class for `Language` ModelAdmin.
+    """
+    def get_linked_pages(self, **kwargs):
+        """Get a list of related translatable pages."""
+        return list(self.instance.pages.all())
+
+    def remove_related_pages(self):
+        """Remove all related translatable pages associated with a Language."""
+        self.instance.pages.all().delete()
+
+    def post(self, request, *args, **kwargs):
+        """Overriding Default post method because we need to delete related
+        translatable pages (if there is any) before we can delete a Language
+        which are references through protected foreign key.
+        """
+        try:
```

Comment:
First check if there are related pages, remove them and finally call the super method which will delete the language object.